src/util/nip04.rs

```diff
@@ -0,0 +1,126 @@
+use aes::Aes256;
+use base64::{decode, encode};
+use block_modes::{block_padding, BlockMode, Cbc};
+use secp256k1::{ecdh, rand::random, schnorrsig, PublicKey, SecretKey};
+use std::convert::From;
+use std::str::FromStr;
+use thiserror::Error;
+
+type Aes256Cbc = Cbc<Aes256, block_padding::Pkcs7>;
+
+#[derive(Error, Debug, PartialEq)]
```

Comment:
I like that you're using thiserror I know it's recommended for when building APIs